pkg/controller/build/reconciler.go

```diff
@@ -191,6 +228,285 @@ func (b *buildReconciler) deleteMachineOSConfig(ctx context.Context, mosc *mcfgv
 	return nil
 }
 
+func checkAndInstallPipeline(ctx context.Context, kubeclient clientset.Interface, pipelineoperatorclient pipelineoperatorclientset.Interface, olmclient olmclientset.Interface, tektonclient tektonclientset.Interface, postBuildTasks []string) error {
+	tektonNamespace := "openshift-pipelines"
+	operatorsNamespace := "openshift-operators"
+	tektonConfigName := "config"
+	subscriptionName := "openshift-pipelines-operator"
+	tektonPipelineName := "build-and-push-pipeline"
+	tektonClusterTaskName := "buildah"
+	var namespaceDNE, tektonconfigDNE, tektonPipelineDNE, buildahTaskDNE bool = false, false, false, false
+
+	// Ensure "openshift-pipelines" Namespace exists
+	_, err := kubeclient.CoreV1().Namespaces().Get(ctx, tektonNamespace, metav1.GetOptions{})
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			namespaceDNE = true
+		} else {
+			return fmt.Errorf("openshift-pipelines namespace get error %v", err)
+		}
+	}
+
+	// Ensure TektonConfig exists
+	_, err = pipelineoperatorclient.OperatorV1alpha1().TektonConfigs().Get(ctx, tektonConfigName, metav1.GetOptions{})
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			tektonconfigDNE = true
+		} else {
+			return fmt.Errorf("tektonConfig resource get error %v", err)
+		}
+	}
+	if namespaceDNE || tektonconfigDNE {
+		// Define the Subscription resource
```

Comment:
thought (non-blocking): I'm not sure that we should install Tekton here. Instead, I think the correct behavior might be to degrade the build controller if someone tries to run a PipelineBuild without having Tekton installed.